Tests/DataLoaderTests/DataLoaderTests.swift

```diff
@@ -1,92 +1,97 @@
     /// Builds a really really simple data loader'
-    func testReallyReallySimpleDataLoader() throws {
-        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        defer {
-            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
-        }
-
+    func testReallyReallySimpleDataLoader() async throws {
         let identityLoader = DataLoader<Int, Int>(
             options: DataLoaderOptions(batchingEnabled: false)
         ) { keys in
-            let results = keys.map { DataLoaderFutureValue.success($0) }
-
-            return eventLoopGroup.next().makeSucceededFuture(results)
+            keys.map { DataLoaderValue.success($0) }
         }
 
-        let value = try identityLoader.load(key: 1, on: eventLoopGroup).wait()
+        let value = try await identityLoader.load(key: 1)
 
         XCTAssertEqual(value, 1)
     }
 
     /// Supports loading multiple keys in one call
-    func testLoadingMultipleKeys() throws {
-        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        defer {
-            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
-        }
-
+    func testLoadingMultipleKeys() async throws {
         let identityLoader = DataLoader<Int, Int>() { keys in
-            let results = keys.map { DataLoaderFutureValue.success($0) }
-
-            return eventLoopGroup.next().makeSucceededFuture(results)
+            keys.map { DataLoaderValue.success($0) }
         }
 
-        let values = try identityLoader.loadMany(keys: [1, 2], on: eventLoopGroup).wait()
+        let values = try await identityLoader.loadMany(keys: [1, 2])
 
         XCTAssertEqual(values, [1, 2])
 
-        let empty = try identityLoader.loadMany(keys: [], on: eventLoopGroup).wait()
+        let empty = try await identityLoader.loadMany(keys: [])
 
         XCTAssertTrue(empty.isEmpty)
     }
 
     // Batches multiple requests
-    func testMultipleRequests() throws {
-        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        defer {
-            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
-        }
-
-        var loadCalls = [[Int]]()
+    func testMultipleRequests() async throws {
+        let loadCalls = Concurrent<[[Int]]>([])
 
         let identityLoader = DataLoader<Int, Int>(
             options: DataLoaderOptions(
                 batchingEnabled: true,
                 executionPeriod: nil
             )
         ) { keys in
-            loadCalls.append(keys)
-            let results = keys.map { DataLoaderFutureValue.success($0) }
+            await loadCalls.mutating { $0.append(keys) }
+
+            return keys.map { DataLoaderValue.success($0) }
+        }
+
+        async let value1 = identityLoader.load(key: 1)
+        async let value2 = identityLoader.load(key: 2)
 
-            return eventLoopGroup.next().makeSucceededFuture(results)
+        try await Task.sleep(nanoseconds: 2_000_000)
```

Comment:
What is the purpose of the hardcoded `Task.sleep` calls in this file? Is there any way we can have the tests without them, in case the test runs on different hardware, etc? If not, could you add comments explaining why this value was chosen, and centralize the `2_000_000` constant into a class-level property?

Reply:
I added a comment to `DataLoaderTests` explaining the purpose of `Task.sleep`.

Reply:
Awesome, thanks for that